### ISSUE TITLE
Naming Conventions for attribute names in ArangoDB

### ISSUE THREAD

**Issue by kevinprotoss** [arangodb/arangodb#1225]
Hi,

I found one limitation in the documentation.

```
Attribute names should not start with the at-mark (@). The at-mark at the start of attribute names is reserved in ArangoDB for future use cases.
```

I need to make sure whether could we use `@` in the attribute names. Certainly, it's supported now in the db. However, if it won't in the future version, that would be a problem.

Why can not use other character for future use cases? As I know `@` in JSON is much more important than other reserve keywords. e.g., JSON-LD uses always attribute names starting with `@`.

Bg,
Kevin

**Comment by pluma** [comment]
I'm not aware of any plans to give special semantics to attributes starting with `@`. I'm not sure whether that line in the documentation is still accurate. Maybe @fceller or @jsteemann could chime in with a more normative opinion?

**Comment by jsteemann** [comment]
I remember we discouraged the usage of `@` in the docs, and we planned to use `@`-prefixed attributes in future versions for special purposes.

That was long ago already, and we did not come up with any concrete plans about such attributes yet. From my point of view, this point of the documentation is outdated and can be changed so any `@`-prefixed attributes can be used, and will not have special meanings in the future. 

However, I'd like @fceller or @neunhoef to confirm before removing this from the docs and encouraging the use of `@`-prefixed attributes.

**Comment by kevinprotoss** [comment]
Thx a lot for your feedback. Then I can use that now.

**Comment by neunhoef** [comment]
I cannot think of a good usage for these attribute names. One question: Would the existence of these attributes pose difficulties in AQL? For example in
  FOR x IN coll
    FILTER x.@name == "Max"
    RETURN x
Would this work? Or would AQL take @name as a query parameter?

**Comment by jsteemann** [comment]
Attribute names can be used outside of AQL, e.g.:
`{"foo":"bar","@name":"bla"}`. In this case, `@name` would be a regular attribute name.
In AQL, a literal `@` can still be used inside an attribute name if the attribute name is enclosed in backticks.

**Comment by fceller** [comment]
I would not encourage the use of '@'

**Comment by jsteemann** [comment]
To clear the confusion that may have arisen, I just discussed that with @fceller: 
- attribute names starting with `@` are supported by ArangoDB and should work, so they can be used by end users if required
- the section about them being reserved for future use cases has been removed from the manual (at least in the source code, the online manual will be updated eventually). 
- we still do not encourage the usage of `@`-prefixed attribute names because in AQL, identifiers starting with the at-mark will be interpreted as bind parameters. To use an at-mark-prefixed attribute name in AQL, it will need to be enclosed in backticks. The same is true for other attribute names by the way, e.g. attribute names including spaces.

Long story short: at-prefixed attributes can be used, though they will need some attention when used in AQL queries.

I think this should provide everyone with enough information to make a good decision about whether it is worth to use such attribute names in their applications.
